src/arch/riscv/kernel/boot.c

```diff
@@ -273,12 +273,12 @@ static BOOT_CODE bool_t try_init_kernel(
     seL4_BootInfoHeader header;
     if (dtb_reg.start) {
         header.id = SEL4_BOOTINFO_HEADER_FDT;
-        header.len = extra_bi_size;
+        header.len = sizeof(header) + dtb_size;
```

Comment:
why recalculate this?

Reply:
It's recalculated here to avoid implicit dependencies, and this is also what the ARM code does. There could be more in `extra_bi_size` than just this, so here I ensure that we just do exactly what we need. Note also, that this is part or a larger patch set, there is more to come where this is wrapper into helper function then that are shared code between ARM and RISC-V. The change here allows better tracing the changes.